src/misc/mntent.c

```diff
@@ -20,6 +22,46 @@ int endmntent(FILE *f)
 	return 1;
 }
 
+static char* decode(char* buf) {
+	assert(buf != NULL);
+	char* src = buf;
+	char* dest = buf;
+	while (1) {
+		char* next_src = __strchrnul(src, '\\');
+		int offset = next_src - src;
```

Comment:
what if next_src is NULL ?

Reply:
If I read the docs right, that shouldn't happen here.

strchr can return NULL, but strchrnul shouldn't. (Assuming the input string ain't NULL.)

Reply:
> The strchr() function returns a pointer to the first occurrence of the character c in the string s.

> The strchrnul() function is like strchr() except that if c is not found in s, then it returns a pointer to the null byte at the end of s, rather than NULL.